packages/bugsnag_flutter/lib/src/client.dart

```diff
@@ -786,8 +787,9 @@ class Bugsnag extends BugsnagClient with DelegateClient {
       'launchDurationMillis': launchDurationMillis,
       'sendLaunchCrashesSynchronously': sendLaunchCrashesSynchronously,
       'appHangThresholdMillis': appHangThresholdMillis,
-      'redactedKeys': List<String>.from(redactedKeys),
-      'discardClasses': List<String>.from(discardClasses),
+      'redactedKeys': List<String>.from(
+          redactedKeys?.map((e) => e.pattern) ?? {'password'}),
+      'discardClasses': List<String>.from(discardClasses.map((e) => e.pattern)),
```

Comment:
This also needs to carry the `RegExp` flags:

| Dart            | Cocoa                                                                                                                                                                                    | Android                                                                                              |
|-----------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------|
| isDotAll        | [NSRegularExpressionDotMatchesLineSeparators](https://developer.apple.com/documentation/foundation/nsregularexpressionoptions/nsregularexpressiondotmatcheslineseparators?language=objc) | [DOTALL](https://developer.android.com/reference/java/util/regex/Pattern#DOTALL)                     |
| isCaseSensitive | [NSRegularExpressionCaseInsensitive](https://developer.apple.com/documentation/foundation/nsregularexpressionoptions/nsregularexpressioncaseinsensitive?language=objc)                   | [CASE_INSENSITIVE](https://developer.android.com/reference/java/util/regex/Pattern#CASE_INSENSITIVE) |
| isMultiLine     | [NSRegularExpressionAnchorsMatchLines](https://developer.apple.com/documentation/foundation/nsregularexpressionoptions/nsregularexpressionanchorsmatchlines?language=objc)               | [MULTILINE](https://developer.android.com/reference/java/util/regex/Pattern#MULTILINE)               |